python/test_downloader.py

```
@@ -2410,6 +2410,32 @@ def test_requires_mp4_compatibility_transcode_for_non_mp4_source(self):
             is True
         )
 
+    def test_requires_mp4_compatibility_transcode_for_incompatible_mp4_codec_probe(self):
+        """AV1 in mp4 should trigger transcode even if container is mp4."""
+        assert (
+            YouTubeDownloader._requires_mp4_compatibility_transcode(
+                Path("/tmp/source.mp4"),
```

Ruff (0.15.1)

[error] Probable insecure usage of temporary file or directory: "/tmp/source.mp4"

(S108)